File: IT/Input-Output/Esercizi/Python/esercizio10.py
# file esercizio10.py

from __future__ import print_function
if hasattr(__builtins__, 'raw_input'): input = raw_input
from os import path, linesep
from datetime import datetime

filename = input("Apri il file: ")

date = datetime.now()

if path.isfile(filename):
	f = open(filename, "a")
	f.write("File modificato il " + date.strftime("%d-%m-%Y alle ore %H:%M:%S") + linesep)
else:
	f = open(filename, "w")
	f.write("File creato il " + date.strftime("%d-%m-%Y alle ore %H:%M:%S") + linesep)

f.close()

File: IT/2/Input-Output/Esercizi/C++/esercizio10.cpp
/*
file esercizio10.cpp
compilare con: g++ -std=c++11 esercizio10.cpp
*/

#include <ctime>
#include <iostream>
#include <fstream>

using std::cin;
using std::cout;
using std::endl;
using std::string;
using std::fstream;

int main()
{
	cout << "Apri il file: ";

	string filename;
	getline(cin, filename);

	fstream f(filename);

	time_t now = time(NULL);
	struct tm *timeinfo = localtime(&now);
	char buffer[80];

	if (f) {
		f.close();
		f.open(filename, fstream::out | fstream::app);
		strftime(buffer, sizeof(buffer), "File modificato il %d-%m-%Y alle ore %H:%M:%S", timeinfo);

		f << string(buffer) << endl;
	}
	else {
		f.open(filename, fstream::out);
		strftime(buffer, sizeof(buffer), "File creato il %d-%m-%Y alle ore %H:%M:%S", timeinfo);

		f << string(buffer) << endl;
	}
}

